src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.put(transaction.getFruit(), storage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
The current implementation does not check if the result of the purchase operation will lead to a negative balance in the storage. According to the checklist, you should check the result balance before saving it in the Storage, and it should be positive. Throw a RuntimeException in case the balance is negative.